### PR TITLE
Only label RePEc's preprints as preprints

### LINKED ISSUE #417: Update RePec normalizer so not everything is categorized as a Preprint
* determine if this is possible based on their data

### PR DESCRIPTION
**Pull request by leb2dg**
Resolves #417 

* this also impacts any OAI creativework that has "preprint" as the type

### CODE REVIEW

**Comment by aaxelb** [comment]
Are there other `dc:type` strings that are commonly used? It'd be awesome if we could recognize Publications and minimize our use of CreativeWork as a catch-all

**Comment by leb2dg** [comment]
`type` isn't a controlled field for OAI but some sources use consistent types so there is some room for improvement.

File: share/normalize/oai.py [hunk 1]
```diff
@@ -109,7 +109,15 @@ class OAIThroughTags(Parser):
...
 class OAICreativeWork(Parser):
-    schema = 'CreativeWork'
+    @property
+    def schema(self):
+        try:
+            resourceType = self.context['record']['metadata']['dc']['dc:type']
+            if resourceType == 'preprint':
+                return 'Preprint'
+            return 'CreativeWork'
+        except KeyError:
```
Comment (chrisseto) [inline share/normalize/oai.py]: You may want to catch `ValueError` here as well just in case one of the key returns `None`

**Comment by leb2dg** [comment]
@chrisseto, comment addressed 🚝

**Comment by chrisseto** [comment]
Merged via https://github.com/CenterForOpenScience/SHARE/commit/6ad28cf3ebb3f1ff7f95603f5f9312a25bd9c376

**Pull request closed without merging**